doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md

```diff
@@ -871,6 +872,62 @@ show rollback log [exec | verify | status]
 
 Use the *verbose* option to view additional details while executing the different commands.
 
+### 3.2.3 Multi ASICs Support
+
+The initial design of the SONiC Generic Configuration Update and Rollback feature primarily focuses on single-ASIC platforms. To cater to the needs of Multi-ASIC platforms, this section introduces enhancements to support configuration updates and rollbacks across multiple ASICs and the host namespace.
+
+#### 3.2.3.1 Overview
+
+In Multi-ASIC SONiC platforms, configurations can be applied either globally, affecting all ASICs, or individually, targeting a specific ASIC or the host. The configuration management tools must, therefore, be capable of identifying and applying configurations based on their intended scope.
+
+#### 3.2.3.2 Namespace-aware Configuration Management
+
+The SONiC utilities for configuration management (apply-patch, checkpoint, and rollback) will be enhanced to become namespace-aware. These utilities will determine the target namespace for each operation from the configuration patch itself or from user input for operations that involve checkpoints and rollbacks.
+
+#### 3.2.3.3 JSON Patch Format Extension
+
+The JSON Patch format will be extended to include the namespace identifier for each operation. Operations that target the host namespace will be marked with a "localhost" identifier, while those intended for a specific ASIC will include an "asicN" identifier, where N denotes the ASIC number.
+
+```json
+
+{
+  "asic0": [
```

Comment:
Not sure if this aligns with json patch diff.
When we generate host and asics config, it is represented as a single json.
If we use json patch to generate the diff, it will show in below format:
```
[
    {"op": "remove", "path": "/asic0/DSCP_TO_TC_MAP"},
    {"op": "add", "path": "/asic0/POLICER", "value": { /* Policer configuration */ }}
    {"op": "add", "path": "/localhost/SYSLOG_SERVER", "value": { /* Syslog server configuration */ }}
 ]
```
May need to compare these two format pro and cons.

Reply:
The pushed jsonpatch will be like 

```json
[
    {"op": "remove", "path": "/asic0/DSCP_TO_TC_MAP"},
    {"op": "add", "path": "/asic0/POLICER", "value": { /* Policer configuration */ }}
    {"op": "add", "path": "/localhost/SYSLOG_SERVER", "value": { /* Syslog server configuration */ }}
 ]
```

But for reuse current single ASIC GCU logic, we need categorize the change per ASIC which includes localhost as well. Then it would be change to like below one as the design proposal:

```json

{
  "asic0": [
    {"op": "remove", "path": "/DSCP_TO_TC_MAP"},
    {"op": "add", "path": "/POLICER", "value": { /* Policer configuration */ }}
  ],
  "localhost": [
    {"op": "add", "path": "/SYSLOG_SERVER", "value": { /* Syslog server configuration */ }}
  ]
}
```

Reply:
LGTM.